pkg/pipeline/CiService.go

```diff
@@ -487,26 +480,20 @@ func (impl *CiServiceImpl) executeCiPipeline(workflowRequest *types.WorkflowRequ
 	return nil
 }
 
-func (impl *CiServiceImpl) buildS3ArtifactLocation(ciWorkflowConfig *pipelineConfig.CiWorkflowConfig, savedWf *pipelineConfig.CiWorkflow) (string, string, string) {
-	ciArtifactLocationFormat := ciWorkflowConfig.CiArtifactLocationFormat
-	if ciArtifactLocationFormat == "" {
-		ciArtifactLocationFormat = impl.config.GetArtifactLocationFormat()
-	}
-	ArtifactLocation := fmt.Sprintf("s3://%s/%s/"+ciArtifactLocationFormat, ciWorkflowConfig.LogsBucket, impl.config.GetDefaultArtifactKeyPrefix(), savedWf.Id, savedWf.Id)
-	artifactFileName := fmt.Sprintf(impl.config.GetDefaultArtifactKeyPrefix()+"/"+ciArtifactLocationFormat, savedWf.Id, savedWf.Id)
-	return ArtifactLocation, ciWorkflowConfig.LogsBucket, artifactFileName
+func (impl *CiServiceImpl) buildS3ArtifactLocation(ciWorkflowConfigLogsBucket string, savedWf *pipelineConfig.CiWorkflow) (string, string, string) {
+	ciArtifactLocationFormat := impl.config.GetArtifactLocationFormat()
+	ArtifactLocation := fmt.Sprintf("s3://"+path.Join(ciWorkflowConfigLogsBucket, ciArtifactLocationFormat), savedWf.Id, savedWf.Id)
```

Comment:
duplicate

Reply:
there are two methods `buildS3ArtifactLocation` and `buildArtifactLocationForS3`. But, both of them serve different business logics.